crates/binstalk/src/helpers/git.rs

```diff
@@ -161,7 +161,7 @@ impl Repository {
                     .to_thread_local()
                     .head_commit()?
                     .tree()?
-                    .lookup_entry_by_path(path)?
+                    .lookup_entry_by_path(path, &mut Vec::new())?
```

Comment:
@Byron Can you double-check on the usage of [`Tree::lookup_entry_by_path`](https://docs.rs/gix/latest/gix/struct.Tree.html#method.lookup_entry_by_path) here?

IMO the `buf` here can just be `&mut Vec::new()` or any `&mut Vec::with_capacity(...)` as it will be reused for storing temporary data?

Reply:
That would be fine, but what's really wanted here is [`peel_to_entry_by_path()`](https://docs.rs/gix/0.50.0/gix/struct.Tree.html#method.peel_to_entry_by_path). It's what it was before, but non-consuming.

Reply:
Thanks!

I think the doc can improve a bit since it is a bit confusing.

Also I tried passing Vec::new() to lookup and it failed in CI.

Reply:
Thanks for the hint! I looked into this and realized that method [wasn't actually working](https://github.com/Byron/gitoxide/commit/145f8658a32c46db1f54d3098cf9371fe6eeec5e) - apologies.

Reply:
> Thanks for the hint! I looked into this and realized that method [wasn't actually working](https://github.com/Byron/gitoxide/commit/145f8658a32c46db1f54d3098cf9371fe6eeec5e) - apologies.

Thank you for the swift response and fix!

I will switch back to peel_to_entry_by_path though I do think it's worth releasing a new patch release.